### DISCUSSION TITLE
Is MetricAverageCallback() necessary when using Horovod on Tensorflow

### DISCUSSION THREAD

**Discussion by unfazing** [horovod/horovod#2795] [Q&A]
**Environment:**
I am using a docker image, horovod/horovod:0.20.0-tf2.3.0-torch1.6.0-mxnet1.6.0.post0-py3.7-cuda10.1, pulled from docker hub on machines running Ubuntu 20.04

**Checklist:**
1. Did you search issues to find if somebody asked this question before? Yes
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.rst)? NIL
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.rst)? NIL
4. Did you check if you question is answered in the [troubleshooting guide] (https://github.com/horovod/horovod/blob/master/docs/troubleshooting.rst)? Yes

**Your question:**
Hello, in the documentation for [Horovod on Tensorflow](https://horovod.readthedocs.io/en/stable/tensorflow.html), the `MetricAverageCallback` is not implemented. 

Seems like it would be necessary to ensure the validation loss is constant for each node such that other functionality like `EarlyStopping` or `ReduceLROnPlateau` can work as expected. (perhaps unless only `hvd.rank==0` runs the validation tests, in which case will the other processes carry on with the next epoch or wait for the validation test to finish on process 0?)

If metric averaging is necessary, are there any examples of its implementation for Horovod on Tensorflow?

**Comment by tgaddair** [comment]
Hey @unfazing, you can find the implementation of the `MetricAverageCallback` [here](https://github.com/horovod/horovod/blob/master/horovod/_keras/callbacks.py#L49). This API is ultimately operating of TensorFlow tensors, so if you wanted to use this outside Keras, it is definitely possible. The only reason it is not shown in the Horovod on TensorFlow API is because this specific implementation is designed to work with Keras' internal metrics API.
